### DISCUSSION TITLE
Nosql Database(Mongodb) support  with bucket4j

### DISCUSSION THREAD

**Discussion by raghu298** [vladimir-bukhtoyarov/bucket4j#394] [Q&A]
Bucket4j is a awesome library compared to all existing libraries like resilence4j.
Just want to know does bucket4j has support with Nosql databases.
If not is there a way to do in nosql databases.
@vladimir-bukhtoyarov Thanks for the library.

**Comment by vladimir-bukhtoyarov** [comment]
All supported persistence technologies are described there https://github.com/bucket4j/bucket4j#non-jvm-back-ends

Mongo DB is not supported. You can do it by yurself via extending [AbstractCompareAndSwapBasedProxyManager](https://github.com/bucket4j/bucket4j/blob/master/bucket4j-core/src/main/java/io/github/bucket4j/distributed/proxy/generic/compare_and_swap/AbstractCompareAndSwapBasedProxyManager.java) or [AbstractLockBasedProxyManager](https://github.com/bucket4j/bucket4j/blob/master/bucket4j-core/src/main/java/io/github/bucket4j/distributed/proxy/generic/pessimistic_locking/AbstractLockBasedProxyManager.java). Integrations [with Redis](https://github.com/bucket4j/bucket4j/tree/master/bucket4j-redis/src/main/java/io/github/bucket4j/redis) and integration [with PostgreSQL](https://github.com/bucket4j/bucket4j/blob/master/bucket4j-postgresql/src/main/java/io/github/bucket4j/postgresql/PostgreSQLadvisoryLockBasedProxyManager.java) can be used as reference.
